mne/time_frequency/psd.py

```diff
@@ -112,7 +106,207 @@ def _check_nfft(n, n_fft, n_overlap):
     return n_fft, n_overlap
 
 
+def _check_psd_data(inst, tmin, tmax, picks, proj):
+    """Helper to do checks on PSD data / pull arrays from inst"""
+    from ..epochs import _BaseEpochs
+    from ..io.base import _BaseRaw
+    from ..evoked import Evoked
```

Comment:
On Tue, Dec 29, 2015 at 12:40 AM, Chris Holdgraf <notifications@github.com>
wrote:

> In mne/time_frequency/psd.py
> <https://github.com/mne-tools/mne-python/pull/2710#discussion_r48498995>:
>
> > @@ -112,7 +106,207 @@ def _check_nfft(n, n_fft, n_overlap):
> >      return n_fft, n_overlap
> >
> >
> > +def _check_psd_data(inst, tmin, tmax, picks, proj):
> > +    """Helper to do checks on PSD data / pull arrays from inst"""
> > +    from ..epochs import _BaseEpochs
> > +    from ..io.base import _BaseRaw
> > +    from ..evoked import Evoked
>
> whenever I do this I get import errors...any idea how to avoid this w/o
> nesting the imports?
>
Oh, I see. It's probably because those functions are also calling this
function somehow and you're getting a circular import. You can leave it
like this I guess.

> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mne-tools/mne-python/pull/2710/files#r48498995>.
>